debezium-core/src/main/java/io/debezium/pipeline/signal/SignalProcessor.java

```diff
@@ -69,14 +72,18 @@ public SignalProcessor(Class<? extends SourceConnector> connector,
                            Offsets<P, O> previousOffsets) {
 
         this.connectorConfig = config;
-        this.signalChannelReaders = signalChannelReaders;
+        this.totalChannelReaders = signalChannelReaders;
         this.documentReader = documentReader;
         this.previousOffsets = previousOffsets;
         this.signalProcessorExecutor = Threads.newSingleThreadScheduledExecutor(connector, config.getLogicalName(), SignalProcessor.class.getSimpleName(), false);
 
-        signalChannelReaders.stream()
+        // filter single channel reader based on configuration
```

Comment:
I am wondering if it's better to extract a method and avoid the comment.

Reply:
Good idea, I will change it.